src/utils.py

```diff
@@ -275,3 +297,103 @@ def upload_experiment_to_s3(experiment_id: str,
     for upload_file in tqdm(upload_files):
         upload_file = str(upload_file)
         s3_client.upload_file(upload_file, bucket_name, upload_file.replace(current_global, object_global))
+
+
+def raven_to_csv_annotations(annotations_path: str,
+                             recording_path: str,
+                             dataset_name: str = 'recordings_2018_filtered'):
+
+    """
+    Takes annotation files (selection table) created in Raven and turns it to a compatible annotations csv.
+    """
+    #create dataframe
+    annotations=Path(annotations_path)
+    recording=Path(recording_path)
+    filelist = list(annotations.glob('*selections.txt')) #ignore irrelevent files
+    metadata = []
+    for file in filelist:
+        dfTemp = pd.read_csv(file, sep="\t")
+        dfTemp['filename'] = re.search(r'18\d{4,4}_\d{6,6}', file.as_posix()).group()
+        dfTemp['StartMicInWater'] = np.amin(dfTemp['Begin Time (s)'])
+        dfTemp['EndMicInWater'] = np.amax(dfTemp['End Time (s)'])
+        metadata.append(dfTemp)
+
+    metadata = pd.concat(metadata)
+    metadata.rename(columns={'Begin Time (s)': 'begin_time', 'End Time (s)': 'end_time'}, inplace=True)
+    print('Number of Labels:', metadata.shape[0])
+    metadata['Annotation'] = metadata['Annotation'].replace(np.nan, 'w', regex=True)
+    #add recording length to dataframe
+    wav_filelist = list(recording.glob('*.wav'))
+    wav_filedict = {re.search(r'18\d{4,4}_\d{6,6}', file.as_posix()).group(): {'path': file} for file in wav_filelist
+                    if re.search(r'18\d{4,4}_\d{6,6}', file.as_posix())}
+    for key, value in wav_filedict.items():
+        record_length = sf.info(value['path']).duration
+        value.update({'length': record_length})
+        wav_filedict[key] = value
+    annotation_lengths = []
+    for filename_txt in metadata['filename']:
+        annotation_lengths.append(wav_filedict[filename_txt]['length'])
+    metadata['TotalRecordLength'] = annotation_lengths
+    # filter metadata to contain only desired call types
+    filters = ['w', 'sc']
+    metadata_calls = metadata[metadata.Annotation.isin(filters)]
+    unique_files, idx = np.unique(metadata['filename'], return_index=True)
+    # Find true length of call sequences ( get rid of over lapping-sequences)
+    non_overlap_all = non_overlap_df(metadata)
+    non_overlap_calls = non_overlap_df(metadata_calls)
+    # label background segments
+    bg_segments = []
+    for file in (unique_files):
+        file_df = non_overlap_all[non_overlap_all['filename'] == file]
+        begin = np.array(file_df['begin_time'])
+        end = np.array(file_df['end_time'])
+        for item in end:
+            next_beginning = begin[begin > item]
+            if next_beginning.size == 0:
+                break
+            next_beginning = np.min(next_beginning)
```

Comment:
next_beginning = next_beginning[0] ? I don't remember if the array is order but I believe it is

Reply:
I just noticed this comment. I didn't understand what you mean.